clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1227,6 +1227,7 @@ public void close(Duration timeout) {
 
     private void close(Duration timeout, boolean swallowException) {
         log.trace("Closing the Kafka consumer");
+        boolean wasInterrupted = Thread.interrupted();
```

Comment:
i wonder if we could just do if(Thread.interrupted()) firstException = new InterruptException()) 

I think if the thread was interrupted, we do want to throw interruptException?

Reply:
> i wonder if we could just do if(Thread.interrupted()) firstException = new InterruptException())

That doesn't really work because while `Thread.interrupted()` clears the flag, the constructor for `InterruptException` calls `Thread.interrupt()` again.

Reply:
you are right
```
    public InterruptException(InterruptedException cause) {
        super(cause);
        Thread.currentThread().interrupt();
    }
```